src/test/java/io/reactivex/internal/operators/flowable/FlowableBufferTest.java

```diff
@@ -2769,4 +2802,15 @@ public void timedSizeBufferAlreadyCleared() {
 
         sub.run();
     }
+
+    @Test
+    public void bufferExactFailingSupplier() {
+        Scheduler.Worker w = new TestScheduler().createWorker();
+        TestSubscriber<List<Integer>> observer = new TestSubscriber<List<Integer>>();
+
+        FlowableBufferTimed.BufferExactBoundedSubscriber<Integer, List<Integer>> buf = new FlowableBufferTimed.BufferExactBoundedSubscriber<Integer, List<Integer>>(observer, ArrayListSupplier.<Integer>asCallable(), 100, TimeUnit.MILLISECONDS, 10, false, w);
```

Comment:
The list supplier should fail, calling onError and onCompleted this way is just a plain protocol violation.

Do something this instead:

```java
Flowable.empty()
.buffer(10, 10, TimeUnit.SECONDS, Schedulers.computation(), new Callable<List<Object>>() {
    @Override public void call() {
        throw new TestException();
    }
})
.test()
.awaitDone(5, TimeUnit.SECONDS)
.assertFailure(TestException.class)
;
```